### PR TITLE
Doctest

### LINKED ISSUE #51: Test documentation examples
It would be nice to have tests to check the documentation examples. This might be achieved via the following: https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html.

### PR DESCRIPTION
**Pull request by james-d-mitchell**
Resolves #51

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
Looks like I missed a bunch of tests in the `src` directory will try to add those now.

File: docs/source/api/matrix.rst [hunk 1]
```diff
@@ -162,10 +164,11 @@ The Matrix class
...
-         .. code-block:: cpp
+         .. doctest::
 
-            // construct a 2 x 3 max-plus truncated matrix
-            m = Matrix(MatrixKind.MaxPlusTrunc, 11, 2, 3);
+            >>> # construct a 2 x 3 max-plus truncated matrix
+            >>> Matrix(MatrixKind.MaxPlusTrunc, 11, 2, 3)
+            Matrix(MatrixKind.MaxPlusTrunc, 11, [[0, 0, 0], [0, 0, 0]])
 
          :raise RunTimeError: if ``kind`` is not :py:attr:`MatrixKind.NTP`.
```
Comment (james-d-mitchell) [inline docs/source/api/matrix.rst:173]:
```suggestion
         :raise RunTimeError: if ``kind`` is not :py:attr:`MatrixKind.NTP`.
```

This doesn't look right.

**Review by MTWhyte — APPROVED** [review]
Looks good! (Modulo the comment you made)

**Comment by james-d-mitchell** [comment]
Thanks @MTWhyte I'll fix the doc typo in a different PR.

**Pull request merged**